java/client/src/main/java/glide/connectors/handlers/ReadHandler.java

```diff
@@ -32,8 +32,20 @@ public void channelRead(@NonNull ChannelHandlerContext ctx, @NonNull Object msg)
     /** Handles uncaught exceptions from {@link #channelRead(ChannelHandlerContext, Object)}. */
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
-        Logger.log(ERROR, "read handler", () -> "=== exceptionCaught " + ctx + " " + cause);
+        if (cause instanceof MessageHandler.MessageCallbackException) {
+            Logger.log(
+                    ERROR,
+                    "read handler",
+                    () -> "=== Exception thrown from pubsub callback " + ctx,
+                    cause.getCause());
 
+            // Mimic the behavior of if this got thrown by a user thread. Print to stderr,
+            cause.printStackTrace();
```

Comment:
why?

Reply:
We're matching the behavior of what happens when you manually create a thread and have an exception escape it.